app/services/preserved_object_handler.rb

```diff
@@ -286,14 +328,16 @@ def confirm_online_version
 
       raise_rollback_if_pc_po_version_mismatch(pres_copy.version, pres_object.current_version)
 
-      if incoming_version == pres_copy.version
+      found_expected_version = (incoming_version == pres_copy.version)
```

Comment:
again, I have a slight preference for `incoming_version == pres_copy.version` in the if statement.  Here, tho, the logic is simpler so my cognitive overload is lower.